tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+        wire_order = [0, 1, 2]
+        mat = sprod_op.matrix(wire_order=wire_order)
+
+        ccnot = math.array(
+            [
+                [1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 1, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1],
+                [0, 0, 0, 0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 0, 1, 0, 0],
+            ]
+        )
+
+        true_mat = scalar * ccnot
+        assert np.allclose(mat, true_mat)
+
+    templates_and_mats = (
+        (qml.QFT(wires=[0, 1, 2]), get_qft_mat(3)),
+        (
+            qml.GroverOperator(wires=[0, 1, 2]),
+            (1 / 4) * math.ones((8, 8), dtype="complex128") - math.eye(8, dtype="complex128"),
+        ),
+    )
+
+    @pytest.mark.parametrize("template, mat", templates_and_mats)
+    def test_sprod_templates(self, template, mat):
+        """Test that we can scale templates and the generated matrix is correct."""
+        scalar = 3.14
+        sprod_op = SProd(scalar, template)
+
+        expected_mat = sprod_op.matrix()
+        true_mat = scalar * mat
+        assert np.allclose(expected_mat, true_mat)
+
+    def test_sprod_qchem_ops(self):
+        """Test that we can scale qchem operations and the generated matrix is correct."""
+        wires = [0, 1, 2, 3]
+        sprod_op1 = SProd(1.23, qml.OrbitalRotation(4.56, wires=wires))
+        sprod_op2 = SProd(3.45, qml.SingleExcitation(1.23, wires=[0, 1]))
+        mat1 = sprod_op1.matrix()
+        mat2 = sprod_op2.matrix()
+
+        or_mat = 1.23 * gd.OrbitalRotation(4.56)
+        se_mat = 3.45 * gd.SingleExcitation(1.23)
+
+        assert np.allclose(mat1, or_mat)
+        assert np.allclose(mat2, se_mat)
+
+    def test_sprod_observables(self):
+        """Test that observable objects can also be scaled with correct matrix representation."""
+        wires = [0, 1]
+        sprod_op1 = SProd(1.23, qml.Projector(basis_state=qnp.array([0, 1]), wires=wires))
+        sprod_op2 = SProd(3.45, qml.Hermitian(qnp.array([[0.0, 1.0], [1.0, 0.0]]), wires=0))
+        mat1 = sprod_op1.matrix()
+        mat2 = sprod_op2.matrix()
+
+        her_mat = 3.45 * qnp.array([[0.0, 1.0], [1.0, 0.0]])
+        proj_mat = 1.23 * qnp.array(
+            [[0.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]]
+        )
+
+        assert np.allclose(mat1, proj_mat)
+        assert np.allclose(mat2, her_mat)
+
+    def test_sprod_qubit_unitary(self):
+        """Test that an arbitrary QubitUnitary can be scaled with correct matrix representation."""
+        U = 1 / qnp.sqrt(2) * qnp.array([[1, 1], [1, -1]])  # Hadamard
+        U_op = qml.QubitUnitary(U, wires=0)
+
+        sprod_op = SProd(42, U_op)
+        mat = sprod_op.matrix()
+
+        true_mat = 42 * U
+        assert np.allclose(mat, true_mat)
```

Comment:
While I am all for over-testing things, I'm not sure we need quite so many tests of different matrices.

Reply:
Probably not, I think we would benefit from keeping this one though, because it tests a different type of operator.